File: docs/data-tools/codesnippet/CSharp/handle-a-concurrency-exception_3.cs
        private string CreateMessage(NorthwindDataSet.CustomersRow cr)
        {
            return
                "Database: " + GetRowData(GetCurrentRowInDB(cr), DataRowVersion.Default) + "\n" +
                "Original: " + GetRowData(cr, DataRowVersion.Original) + "\n" +
                "Proposed: " + GetRowData(cr, DataRowVersion.Current) + "\n" +
                "Do you still want to update the database with the proposed value?";
        }


        //--------------------------------------------------------------------------
        // This method loads a temporary table with current records from the database
        // and returns the current values from the row that caused the exception.
        //--------------------------------------------------------------------------
        private NorthwindDataSet.CustomersDataTable tempCustomersDataTable = 
            new NorthwindDataSet.CustomersDataTable();

        private NorthwindDataSet.CustomersRow GetCurrentRowInDB(NorthwindDataSet.CustomersRow RowWithError)
        {
            this.customersTableAdapter.Fill(tempCustomersDataTable);

            NorthwindDataSet.CustomersRow currentRowInDb = 
                tempCustomersDataTable.FindByCustomerID(RowWithError.CustomerID);

            return currentRowInDb;
        }


        //--------------------------------------------------------------------------
        // This method takes a CustomersRow and RowVersion 
        // and returns a string of column values to display to the user.
        //--------------------------------------------------------------------------
        private string GetRowData(NorthwindDataSet.CustomersRow custRow, DataRowVersion RowVersion)
        {
            string rowData = "";

            for (int i = 0; i < custRow.ItemArray.Length ; i++ )
            {
                rowData = rowData + custRow[i, RowVersion].ToString() + " ";
            }
            return rowData;
        }

File: docs/data-tools/codesnippet/VisualBasic/handle-a-concurrency-exception_3.vb
    Private Function CreateMessage(ByVal cr As NorthwindDataSet.CustomersRow) As String
        Return "Database: " & GetRowData(GetCurrentRowInDB(cr), 
                                         Data.DataRowVersion.Default) & vbCrLf &
               "Original: " & GetRowData(cr, Data.DataRowVersion.Original) & vbCrLf &
               "Proposed: " & GetRowData(cr, Data.DataRowVersion.Current) & vbCrLf &
               "Do you still want to update the database with the proposed value?"
    End Function


    '--------------------------------------------------------------------------
    ' This method loads a temporary table with current records from the database
    ' and returns the current values from the row that caused the exception.
    '--------------------------------------------------------------------------
    Private TempCustomersDataTable As New NorthwindDataSet.CustomersDataTable

    Private Function GetCurrentRowInDB(
        ByVal RowWithError As NorthwindDataSet.CustomersRow
        ) As NorthwindDataSet.CustomersRow

        Me.CustomersTableAdapter.Fill(TempCustomersDataTable)

        Dim currentRowInDb As NorthwindDataSet.CustomersRow =
            TempCustomersDataTable.FindByCustomerID(RowWithError.CustomerID)

        Return currentRowInDb
    End Function


    '--------------------------------------------------------------------------
    ' This method takes a CustomersRow and RowVersion 
    ' and returns a string of column values to display to the user.
    '--------------------------------------------------------------------------
    Private Function GetRowData(ByVal custRow As NorthwindDataSet.CustomersRow,
        ByVal RowVersion As Data.DataRowVersion) As String

        Dim rowData As String = ""

        For i As Integer = 0 To custRow.ItemArray.Length - 1
            rowData &= custRow.Item(i, RowVersion).ToString() & " "
        Next

        Return rowData
    End Function